File: app/src/main/java/tw/y_studio/ptt/Fragment/EmptyFragment.java
package tw.y_studio.ptt.Fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import tw.y_studio.ptt.Adapter.SearchBoardsAdapter;
import tw.y_studio.ptt.R;
import tw.y_studio.ptt.UI.BaseFragment;
import tw.y_studio.ptt.UI.CustomLinearLayoutManager;

public class EmptyFragment extends BaseFragment {

    public static EmptyFragment newInstance() {
        Bundle args = new Bundle();
        EmptyFragment fragment = new EmptyFragment();
        fragment.setArguments(args);
        return fragment;
    }

    public static EmptyFragment newInstance(Bundle args) {
        EmptyFragment fragment = new EmptyFragment();
        fragment.setArguments(args);
        return fragment;
    }

    private RecyclerView recyclerView;
    private SearchBoardsAdapter mdapter;
    private List<Map<String, Object>> data = new ArrayList<>();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.empty_fragment_layout, container, false);

        setMainView(view);

        recyclerView = findViewById(R.id.recyclerView_empty);

        Bundle bundle = getArguments();//取得Bundle

        mdapter = new SearchBoardsAdapter(getCurrentActivity(),data);

        final CustomLinearLayoutManager layoutManager = new CustomLinearLayoutManager(getContext());
        layoutManager.setOrientation(RecyclerView.VERTICAL);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(mdapter);


        return view;
    }

    protected void onAnimOver() {

    }

}

File: app/src/main/java/cc/ptt/android/presentation/home/EmptyFragment.kt
package cc.ptt.android.presentation.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import cc.ptt.android.R
import cc.ptt.android.presentation.base.BaseFragment
import cc.ptt.android.presentation.common.CustomLinearLayoutManager

class EmptyFragment : BaseFragment() {
    private var recyclerView: RecyclerView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.empty_fragment_layout, container, false)
        setMainView(view)
        recyclerView = findViewById<RecyclerView>(R.id.recyclerView_empty)
        val layoutManager = CustomLinearLayoutManager(context)
        layoutManager.orientation = RecyclerView.VERTICAL
        recyclerView!!.setHasFixedSize(true)
        recyclerView!!.layoutManager = layoutManager
        recyclerView!!.adapter = object : RecyclerView.Adapter<RecyclerView.ViewHolder?>() {
            override fun getItemCount(): Int {
                return 0
            }

            override fun onCreateViewHolder(
                parent: ViewGroup,
                viewType: Int
            ): RecyclerView.ViewHolder {
                return object : RecyclerView.ViewHolder(View(null)) {
                }
            }

            override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
            }
        }
        return view
    }

    override fun onAnimOver() {}

    companion object {
        @JvmStatic
        fun newInstance(): EmptyFragment {
            val args = Bundle()
            val fragment = EmptyFragment()
            fragment.arguments = args
            return fragment
        }

        @JvmStatic
        fun newInstance(args: Bundle?): EmptyFragment {
            val fragment = EmptyFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
